consensus/state_sync/module.go

```diff
@@ -12,47 +14,50 @@ const (
 	stateSyncModuleName = "stateSyncModule"
 )
 
-type SyncMode string
-
-const (
-	Sync      SyncMode = "sync"
-	Synched   SyncMode = "synched"
-	Pacemaker SyncMode = "pacemaker"
-	Server    SyncMode = "server"
-)
-
 type StateSyncModule interface {
 	modules.Module
 	StateSyncServerModule
+	DebugStateSync
 
-	// Handle a metadata response from a peer so this node can update its local view of the state
-	// sync metadata available from its peers
-	HandleStateSyncMetadataResponse(*typesCons.StateSyncMetadataResponse) error
+	IsServerModEnabled() bool
+	EnableServerMode() error
+	DisableServerMode() error
 
-	// Handle a block response from a peer so this node can update apply it to its local state
-	// and catch up to the global world state
+	HandleStateSyncMetadataResponse(*typesCons.StateSyncMetadataResponse) error
 	HandleGetBlockResponse(*typesCons.GetBlockResponse) error
 
-	IsServerModEnabled() bool
-	EnableServerMode()
-
 	SendStateSyncMessage(msg *typesCons.StateSyncMessage, nodeAddress cryptoPocket.Address, height uint64) error
+
+	GetSyncMetadataBuffer() []*typesCons.StateSyncMetadataResponse
```

Comment:
Do we need to expose both functions? I was thinking that one could (potentially) be enough?

I'm still looking through the rest of the code though

Reply:
I removed `GetSyncMetadataBuffer()` as issue #352 doesn't need it anymore, so that's right.